Sources/RIDIFoundation/XML/_XMLParser.swift

```diff
@@ -0,0 +1,108 @@
+        xmlParser = .init(data: data)
+        super.init()
+
+        xmlParser.delegate = self
+    }
+
+    override func main() {
+        result = nil
+        xmlParser.parse()
+        result = result ?? xmlParser.parserError.flatMap { .failure($0) }
+    }
+}
+
+extension _XMLParser: XMLParserDelegate {
+    func parserDidStartDocument(_ parser: XMLParser) {
+        xmlDocument = XMLDocument()
+        xmlDocumentCurrentIndexPath = IndexPath()
+    }
+
+    func parser(_ parser: XMLParser, parseErrorOccurred parseError: Error) {
+        result = .failure(parseError)
+    }
+
+    func parser(_ parser: XMLParser, validationErrorOccurred validationError: Error) {
+        result = .failure(validationError)
+    }
+
+    func parserDidEndDocument(_ parser: XMLParser) {
+        result = .success(xmlDocument)
+    }
+
+    func parser(
+        _ parser: XMLParser,
+        didStartElement elementName: String,
+        namespaceURI: String?,
+        qualifiedName qName: String?,
+        attributes attributeDict: [String: String] = [:]
+    ) {
+        let newElement = XMLElement()
+        newElement.name = elementName
+        newElement.attributes = attributeDict.map {
+            let attribute = XMLAttribute()
+            attribute.name = $0.key
+            attribute.stringValue = $0.value
+
+            return attribute
+        }
+
+        isNodeOpened = true
+
+        guard !xmlDocumentCurrentIndexPath.isEmpty else {
+            xmlDocument.addChild(newElement)
+            xmlDocumentCurrentIndexPath.append(xmlDocument.children!.endIndex - 1)
+            return
+        }
+
+        xmlDocument[xmlDocumentCurrentIndexPath]!.addChild(newElement)
+        xmlDocumentCurrentIndexPath.append((xmlDocument[xmlDocumentCurrentIndexPath]?.children?.endIndex ?? 1) - 1)
+    }
+
+    func parser(
+        _ parser: XMLParser,
+        didEndElement elementName: String,
+        namespaceURI: String?,
+        qualifiedName qName: String?
+    ) {
+        isNodeOpened = false
+
+        var parentElementIndexPath = xmlDocumentCurrentIndexPath
+        repeat {
+            guard xmlDocument[xmlDocumentCurrentIndexPath]?.name == elementName else {
+                parentElementIndexPath.removeLast()
+                continue
+            }
+
+            xmlDocumentCurrentIndexPath.removeLast(xmlDocumentCurrentIndexPath.count - parentElementIndexPath.count + 1)
+            break
+        } while (!parentElementIndexPath.isEmpty)
+    }
+
+    func parser(_ parser: XMLParser, foundCharacters string: String) {
+        guard
+            xmlDocument[xmlDocumentCurrentIndexPath] != nil,
+            isNodeOpened
+        else {
+            return
+        }
+
+        if xmlDocument[xmlDocumentCurrentIndexPath]!.stringValue != nil {
+            xmlDocument[xmlDocumentCurrentIndexPath]!.stringValue!.append(string)
+        } else {
+            xmlDocument[xmlDocumentCurrentIndexPath]!.stringValue = string
+        }
+    }
```

Comment:
xmlDocument가 class라 아래 처럼 구현해도 문제 없을 것 같습니다. 
```suggestion
        guard
            let currentIndexpath = xmlDocument[xmlDocumentCurrentIndexPath],
            isNodeOpened
        else {
            return
        }

        if currentIndexpath.stringValue != nil {
            currentIndexpath.stringValue!.append(string)
        } else {
            currentIndexpath.stringValue = string
        }
    }
```

Reply:
Fixed by f749166b356f712e538d3fcb1f9ca98e5a5b1d18